client/llb/exec.go

```diff
@@ -270,6 +270,10 @@ func (e *ExecOp) Marshal(c *Constraints) (digest.Digest, []byte, *pb.OpMetadata,
 		addCap(&e.constraints, pb.CapMountSecret)
 	}
 
+	if len(e.ssh) > 0 {
+		addCap(&e.constraints, pb.CapMountSSH)
```

Comment:
As discovered by @ijc , these caps (same for secrets) need to be added before the constraints are marshaled.